it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/storage/GcsResourceManager.java

```diff
@@ -75,9 +81,21 @@ public final class GcsResourceManager implements ArtifactClient, ResourceManager
 
   public GcsResourceManager(Builder builder) {
     this.client = ArtifactUtils.createStorageClient(builder.credentials);
-    this.bucket = builder.bucket;
     this.testClassName = builder.testClassName;
     this.runId = ArtifactUtils.createRunId();
+    if (builder.bucket == null || builder.bucket.isBlank()) {
```

Comment:
Is it possible to cover this in tests? It looks like the whole if/else block isn't covered.